migrate SDT/Resistance fields out of mob_family_system.sql into new table, hook up fields in core

TeoTwawki opened the pull request:
<!-- remove space and place 'x' mark between square [] brackets or click the checkbox after saving to affirm: -->
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

TeoTwawki committed:
migrate SDT/Resistance fields out of mob_family_system.sql into new table. For now this is cloned from the family ID's so they match, but they are no longer tied to family. 2 mobs in the same family can be given difference SDT/Resistance from here out.

I have separated SDT from Resistance. Even if retail pairs these effects (I believe wiki terminology is confused and this is not the case),
 we need separate modifiers to accurately represent the changes - one value will not fit both effects under our core, and we -want- to be able to adjust them separately anyway rather than code for edge case exceptions.

Item mods may need correction - a lot of past contributions in old DSP used the mods interchangeable, and at several points in project history these were flipped to be the opposite of what they were. This means that for everything I fixed, I am positive I also broke things. Followup commits to handle that aspect coming soon.

zach2good commented on src/map/instance_loader.cpp:
```diff
@@ -186,48 +188,57 @@ CInstance* CInstanceLoader::LoadInstance(CInstance* instance)
-            PMob->m_Element     = (uint8)Sql_GetIntData(SqlInstanceHandle, 47);
-            PMob->m_Family      = (uint16)Sql_GetIntData(SqlInstanceHandle, 48);
-            PMob->m_name_prefix = (uint8)Sql_GetIntData(SqlInstanceHandle, 49);
-            PMob->m_flags       = (uint32)Sql_GetIntData(SqlInstanceHandle, 50);
+            PMob->attRank = (uint8)Sql_GetIntData(SqlInstanceHandle, 35);
```
Thanks for reorganising this nightmare!

TeoTwawki replied:
twice even. I caught it the 1st time, lost it rebasing, and then spent hrs not realizing why all my stuff was broken.

TeoTwawki committed:
Realign scripts to match new naming conventions of resistance and SDT modifiers

TeoTwawki commented:
have now tested. working as expected - just needs item mods corrected now. Hold merge till that is done.

Xaver-DaRed commented on scripts/globals/items/per._snow_cone.lua:
```diff
@@ -27,14 +27,14 @@ item_object.onEffectGain = function(target, effect)
     target:delMod(xi.mod.FOOD_MP_CAP, 50)
     target:delMod(xi.mod.INT, 3)
-    target:delMod(xi.mod.WINDDEF, 5)
+    target:delMod(xi.mod.WIND_SDT, 5)
```
Should be RES. Same for the one above in same file.

Xaver-DaRed commented on src/map/utils/petutils.cpp:
```diff
@@ -898,28 +900,28 @@ namespace petutils
+
+        PPet->setModifier(Mod::FIRE_SDT, petData->fire_sdt);       // These are stored as floating percentages
+        PPet->setModifier(Mod::ICE_SDT, petData->ice_sdt);         // and need to be adjusted into modifier units.
+        PPet->setModifier(Mod::WIND_SDT, petData->wind_sdt);       // Higher SDT = lower damage.
```
That comment doesnt make sense?
Higher Res does mean lower damage, for A or B reason, but higher Specific Damage Taken... Should mean higer damage?

TeoTwawki replied:
in the sql end, its a % basically, 0.00 would mean no dmg, and 1.00 would mean full damage

but in the modifier is higher value = less damage, its inverted and stupid but I have not changed it yet - when the mod gets changed to be 1000 base instead of 256 that'll happen and the comment will be saner. was just to much to explain in line so its more a self reminder wtf the mod does.

edit: the physical ones (slash, h2h etc) already work the way I want to make the elemental ones work. I just lack time to make sure all the scripts are on that method right now and want to get the table changes in

TeoTwawki committed:
fix wrong modifier in a snow cone

TeoTwawki committed:
Realign item_mods.sql to the the changes in resistance mods.

Note: I started to comment the lines, as 90% of them had nothing to guide would be reviewers or future editors on wtf is there, but there are just so many this isn't feasible. My OCD is screaming at me for giving up but I am all out of steam and my fuckthisitus has won out. Something needs to happen to make simply re-dumping the table more palatable (losing all comments makes re-dumping painful, and is why most rows have none)

TeoTwawki committed:
Replace a comment

TeoTwawki committed:
Realign augments and item_latents tables to use corrected resistance modifier IDs too.. Checked pet tables, none affected.

TeoTwawki committed:
typo
